### PR TITLE
Chore/rust bitcoin preparation

### LINKED ISSUE #9: Move crate into rust-bitcoin
Hi,

Some things to consider for this to be brought over into rust-bitcoin. These are my observations only, I do not speak authoritatively for the rust-bitcoin project. Also, forgive me if these things have been mentioned/discussed already.

- Consider fleshing out the manifest i.e., make the crate ready for release on crates.io
- Add a README.md file (mention MSRV)
- Add a CHANGELOG.md file
- Copy the `rustfmt.toml` from `rust-bitcoin` and run `cargo +nightly fmt`
- Copy the `clippy.toml` file from `rust-bitcoin` to configure `clippy` to use the MSRV 

### Bonus Points

- Improve rustdoc form (use conventional sections, headings, tense etc.)
- Use same coding convention configurations as `rust-bitcoin` (included below) and fix all `clippy` warnings introduced by these
- Use `#[cfg_attr(docsrs, doc(cfg(feature = "rand)))]` anywhere that uses `#[cfg(feature = "rand")]` (adds a little badge to docs saying requires feature X, saves explicitly mentioning this in rustdoc)

```rust
// Coding conventions.
#![deny(non_upper_case_globals)]
#![deny(non_camel_case_types)]
#![deny(non_snake_case)]
#![deny(unused_mut)]
#![deny(missing_docs)]
```

(Please note I did not study the logic closely, these comments are all just surface things).

Thanks man!

### PR DESCRIPTION
**Pull request by Tibo-lg**
Fixes #9

### CODE REVIEW

**Comment by yancyribbens** [comment]
Hey @Tibo-lg - thanks for adding this PR.  I think there is still some work to make the doc comments legible.  For example, if you run the following command from this branch:

`cargo docs --open`

you'll notice it needs some updates.  I guess since you started this PR it makes sense to finish it, otherwise I can add to this PR to fix the doc comments.

File: Cargo.toml [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
-authors = ["Crypto Garage"]
+authors = [
+  "Crypto Garage",
+  "Yancy Ribbens <email@yancy.lol>",
+]
 edition = "2018"
+homepage = "https://github.com/p2pderivatives/rust-bitcoin-coin-selection/"
+license = "CC0-1.0"
 name = "rust-bitcoin-coin-selection"
+repository = "https://github.com/p2pderivatives/rust-bitcoin-coin-selection/"
```
Comment (yancyribbens) [inline Cargo.toml:10]: If this is to move to Rust Bitcoin, then I think the repository will be relocated.  Is there any problem with changing this to `"https://github.com/rust-bitcoin/coin-selection/"` with this PR?

**Comment by yancyribbens** [comment]
Fixed the doc comment compile warning in https://github.com/p2pderivatives/rust-bitcoin-coin-selection/pull/11.  Disregard my previous comment.

File: rustfmt.toml [hunk 2]
```diff
@@ -0,0 +1,87 @@
+# Eventually this shoud be: ignore = []
+ignore = [
+       "bitcoin/src/blockdata",
+       "bitcoin/src/consensus",
+       "bitcoin/src/crypto",
+       "bitcoin/src/psbt",
+       "bitcoin/src/util",
+       "hashes",
```
Comment (tcharding) [inline rustfmt.toml:8]: These can all be removed :)

**Review by tcharding — APPROVED** [review]
LGTM, thanks man. BTW I love the PR title, made me chuckle. There are some unneeded things in the github action but its plenty good enough for now. Thanks again.

File: Cargo.toml (hunk 1 above)
Comment (Tibo-lg) [reply Cargo.toml]: Mmmm it makes for a broken link but I updated it anyway.

File: rustfmt.toml (hunk 2 above)
Comment (Tibo-lg) [reply rustfmt.toml]: Thanks fixed!

**Comment by Tibo-lg** [comment]
@yancyribbens please have a last look I think I fixed everything

**Comment by yancyribbens** [comment]
@Tibo-lg these changes look good to me.

**Comment by yancyribbens** [comment]
Heh reading through this is a blast from the past.  I realized recently that I actually borked the compilation of the project by not maintaining these feature gates.  Fixed here https://github.com/p2pderivatives/rust-bitcoin-coin-selection/pull/170

Cheers

**Pull request merged**